File: app/src/main/java/ir/mymessage/service/ApiInterface.java
package ir.mymessage.service;

import java.util.ArrayList;

import ir.mymessage.model.remote.Friend;
import ir.mymessage.model.remote.Message;
import ir.mymessage.model.response.DialogsResponse;
import ir.mymessage.model.response.FriendsResponse;
import ir.mymessage.model.response.LoginResponse;
import ir.mymessage.model.remote.User;
import ir.mymessage.model.response.MessagesResponse;
import ir.mymessage.model.response.SendMessageResponse;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface ApiInterface {

    /*@POST("dialogs")
    Call<ArrayList<DialogsResponse>> dialogs(@Query("user_id") String str);*/

    @POST("login")
    Call<LoginResponse> login(@Body User user);
    @POST("signup")
    Call<LoginResponse> signup(@Body User user);

    @POST("messages")
    Call<ArrayList<MessagesResponse>> messages(@Query("friend_id") String str);

    @POST("message")
    Call<SendMessageResponse> sendMessage(@Body Message message);

    @POST("push-notification")
    Call<String> pushMessage(@Body Message message);

    @POST("allfriends")
    Call<ArrayList<DialogsResponse>> dialogs(@Query("user_id") String str);
    @POST("all-friends")
    Call<ArrayList<FriendsResponse>> allFriends(@Query("user_id") String str);
    @POST("add-friend")
    Call<String> addFriend(@Body Friend friend);
    @POST("delete-friend")
    Call<String> deleteFriend(@Body Friend friend);
    @POST("search-friend")
    Call<User> searchFriend(@Query("username") String str);

    @POST("updatefcmtoken")
    Call<String> updateFcmToken(@Query("user_id") String str, @Query("fcm_token") String str2);
}

File: app/src/main/java/ir/mymessage/service/ApiInterface.kt
package ir.mymessage.service

import java.util.ArrayList

import ir.mymessage.model.remote.Friend
import ir.mymessage.model.remote.Message
import ir.mymessage.model.response.DialogsResponse
import ir.mymessage.model.response.FriendsResponse
import ir.mymessage.model.response.LoginResponse
import ir.mymessage.model.remote.User
import ir.mymessage.model.response.MessagesResponse
import ir.mymessage.model.response.SendMessageResponse
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.Query

interface ApiInterface {

    @POST("login")
    fun login(@Body user: User): Call<LoginResponse>

    @POST("signup")
    fun signup(@Body user: User): Call<LoginResponse>

    @POST("messages")
    fun messages(@Query("friend_id") str: String): Call<ArrayList<MessagesResponse>>

    @POST("message")
    fun sendMessage(@Body message: Message): Call<SendMessageResponse>

    @POST("push-notification")
    fun pushMessage(@Body message: Message): Call<String>

    @POST("allfriends")
    fun dialogs(@Query("user_id") str: String): Call<ArrayList<DialogsResponse>>

    @POST("all-friends")
    fun allFriends(@Query("user_id") str: String): Call<ArrayList<FriendsResponse>>

    @POST("add-friend")
    fun addFriend(@Body friend: Friend): Call<String>

    @POST("delete-friend")
    fun deleteFriend(@Body friend: Friend): Call<String>

    @POST("search-friend")
    fun searchFriend(@Query("username") str: String): Call<User>

    @POST("updatefcmtoken")
    fun updateFcmToken(@Query("user_id") str: String, @Query("fcm_token") str2: String): Call<String>
}
